libcontainer/nsenter/nsexec.c

```diff
@@ -413,60 +440,310 @@ void nsexec(void)
 
 	/* clone(2) flags are mandatory. */
 	if (config.cloneflags == -1)
-		bail("missing clone_flags");
+		bail("missing cloneflags");
 
 	/* Pipe so we can tell the child when we've finished setting up. */
-	if (pipe(syncpipe) < 0)
+	if (socketpair(AF_LOCAL, SOCK_STREAM, 0, syncpipe) < 0)
 		bail("failed to setup sync pipe between parent and child");
 
-	/* Set up the jump point. */
-	if (setjmp(env) == JUMP_VAL) {
-		/*
-		 * We're inside the child now, having jumped from the
-		 * start_child() code after forking in the parent.
-		 */
-		uint8_t s = 0;
-		int consolefd = config.consolefd;
+	/* TODO: Currently we aren't dealing with child deaths properly. */
+
+	/*
+	 * Okay, so this is quite annoying.
+	 *
+	 * In order to make sure that deal with older kernels (when CLONE_NEWUSER
```

Comment:
Yeah, you're right. The actual reason for the split is the various issues described underneath the section. This split is the **whole reason** for this PR (so removing the split would make this PR kinda pointless), as there are several legitimate non-legacy issues fixed by this split such as #975.

I can fix the comment, but I don't really want to wait another two weeks for two LGTMs because people are quite busy. I can fix the comment in a later PR if it's not a huge issue for you right now.